q2_quality_filter/tests/test_filter.py

```diff
@@ -0,0 +1,750 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2024, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pandas as pd
+import pandas.testing as pdt
+
+from copy import copy
+import gzip
+import os
+from pathlib import Path
+import tempfile
+import unittest
+
+import qiime2
+from qiime2.sdk import Artifact
+from qiime2.plugin.testing import TestPluginBase
+from qiime2.util import redirected_stdio
+from q2_types.sample_data import SampleData
+from q2_types.per_sample_sequences import (
+    FastqGzFormat,
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt,
+    PairedEndSequencesWithQuality,
+)
+
+from q2_quality_filter._filter import (
+    FastqRecord,
+    _read_fastq_records,
+    _find_low_quality_window,
+    _truncate,
+    RecordStatus,
+    _process_record,
+    _is_retained,
+    _write_record,
+)
+from q2_quality_filter._format import QualityFilterStatsFmt
+
+
+class HelperMethodTests(TestPluginBase):
+    package = 'q2_quality_filter.tests'
+
+    def test_read_fastq_records(self):
+        exp = [
+            FastqRecord(b'@foo', b'ATGC', b'+', b'IIII'),
+            FastqRecord(b'@bar', b'TGCA', b'+', b'ABCD')
+        ]
+
+        obs = list(
+            _read_fastq_records(self.get_data_path('simple.fastq.gz'))
+        )
+
+        self.assertEqual(len(obs), 2)
+
+        attrs = [
+            'sequence_header', 'sequence', 'quality_header', 'quality_scores'
+        ]
+        for exp_record, obs_record in zip(exp, obs):
+            for attr in attrs:
+                self.assertEqual(
+                    exp_record.__getattribute__(attr),
+                    obs_record.__getattribute__(attr)
+                )
+
+    def test_find_low_quality_window(self):
+        # test no low quality window returns none
+        # 'M' has quality score of 44 with PHRED offset of 33
+        quality_scores = b'M' * 10
+        obs = _find_low_quality_window(
+            quality_scores, phred_offset=33, min_quality=20, window_length=2
+        )
+        self.assertEqual(obs, None)
+
+        # test that `min_quality` bases are not considered part of a window
+        # (only scores that are lower)
+        obs = _find_low_quality_window(
+            quality_scores, phred_offset=33, min_quality=44, window_length=2
+        )
+
```

Comment:
You're missing the assertion here. 
```suggestion
self.assertEqual(obs, None)
```